src/SFA.DAS.RoatpAssessor/SFA.DAS.RoatpAssessor.Web/Validators/RoatpAssessorPageValidator.cs

```diff
@@ -0,0 +1,89 @@
+﻿using SFA.DAS.RoatpAssessor.Web.ApplyTypes;
+using SFA.DAS.RoatpAssessor.Web.ApplyTypes.Validation;
+using SFA.DAS.RoatpAssessor.Web.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.RoatpAssessor.Web.Validators
+{
+    public class RoatpAssessorPageValidator : IRoatpAssessorPageValidator
+    {
+        private const int MaxWordAcount = 150;
+        private const string FailDetailsRequired = "Enter comments";
+        private const string TooManyWords = "Your comments must be 150 words or less";
+        private const string NoSelectionErrorMessage = "Select an outcome";
+
+        public async Task<ValidationResponse> Validate(SubmitAssessorPageAnswerCommand command)
+        {
+            var validationResponse = new ValidationResponse
+            {
+                Errors = new List<ValidationErrorDetail>()
+            };
+
+            if (string.IsNullOrWhiteSpace(command.Status))
+            {
+                validationResponse.Errors.Add(new ValidationErrorDetail("OptionPass", NoSelectionErrorMessage));
+            }
+            else
+            {
+                if (command.Status == AssessorPageReviewStatus.Fail && string.IsNullOrEmpty(command.OptionFailText))
+                {
+                    validationResponse.Errors.Add(new ValidationErrorDetail("OptionFailText",
+                        FailDetailsRequired));
+                }
+            }
+
+            if (validationResponse.Errors.Any())
+            {
+                return await Task.FromResult(validationResponse);
+            }
+
+            switch (command.Status)
+            {
+                case AssessorPageReviewStatus.Pass when !string.IsNullOrEmpty(command.OptionPassText):
+                    {
+                        var wordCount = command.OptionPassText.Split(new[] { " " }, StringSplitOptions.RemoveEmptyEntries).Length;
```

Comment:
repeating 'command.OptionPassText.Split(new[] { " " }, StringSplitOptions.RemoveEmptyEntries).Length;' convert to an internal function

Reply:
sorted